openstf_dbc/services/model_input.py

```diff
@@ -55,110 +56,34 @@ def get_model_input(
 
-        # Get load data
+        # Get load
         load = Ems().get_load_pid(
             pid, datetime_start, datetime_end, forecast_resolution
         )
-        if len(load) == 0:
-            self.logger.warning("Length of load data was 0")
-
-        # Get APX price data
-        apx_data = Predictor().get_apx(datetime_start, datetime_end)
-
-        # Get gas price data
-        gas_data = Predictor().get_gas_price(datetime_start, datetime_end)
-
-        # Get SJV data
-        sjv_data = Predictor().get_tdcv_load_profiles(datetime_start, datetime_end)
-
-        # Get weather data
-        weather_params = [
-            "clouds",
-            "radiation",
-            "temp",
-            "winddeg",
-            "windspeed",
-            "windspeed_100m",
-            "pressure",
-            "humidity",
-            "rain",
-            "mxlD",
-            "snowDepth",
-            "clearSky_ulf",
-            "clearSky_dlf",
-            "ssrunoff",
-        ]
-
-        weather_data = Weather().get_weather_data(
-            location,
-            weather_params,
-            datetime_start,
-            datetime_end,
-            source="optimum",
+        load = load.resample(forecast_resolution).mean().interpolate(limit=3)
+        # Get predictors
+        predictors = Predictor().get_predictors(
+            datetime_start=datetime_start,
+            datetime_end=datetime_end,
+            forecast_resolution=forecast_resolution,
+            location=location,
         )
-
-        # Post process weather data
-        if "source_1" in list(weather_data):
-            weather_data["source"] = weather_data.source_1
-            weather_data = weather_data.drop("source_1", axis=1)
-        if "input_city_1" in list(weather_data):
-            del weather_data["input_city_1"]
-        else:
-            del weather_data["input_city"]
-        del weather_data["source"]
-
-        # Combine data
-        result = (
-            pd.DataFrame(
-                index=pd.date_range(
-                    start=datetime_start,
-                    end=datetime_end,
-                    freq=forecast_resolution,
-                    tz="UTC",
-                )
-            )
-            .resample(forecast_resolution)
-            .ffill()
+        # create model input with datetime index
+        model_input = pd.DataFrame(
+            index=get_datetime_index(datetime_start, datetime_end, forecast_resolution)
         )
-        result.index.name = "index"
-
-        # Fill return dataframe with all data collected
-        if len(load) > 0:
-            result = pd.concat(
-                [
-                    result,
-                    load.resample(forecast_resolution).mean().interpolate(limit=3),
-                ],
-                axis=1,
-            )
+        model_input.index.name = "index"
+
+        # Add load
+        if load.empty is False:
```

Comment:
`if not load.empty:`

Reply:
Both are fine by me. `is False` is a bit more explicit many things are not True (like 0, empty list, empty string, None etc.) but things that are not True are not equal to the instance of the boolean False. Anyway, I can change it of course!

Reply:
changed!